src/Adapter/Image/ProductImageValidator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Image;
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Adapter\File\ByteConvertor;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedProductImageConstraintException;
+
+/**
+ * Responsible for validating image before upload
+ */
+class ProductImageValidator
+{
+    /**
+     * @var ImageValidator
+     */
+    private $imageValidator;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    public function __construct(ImageValidator $imageValidator, Configuration $configuration)
+    {
+        $this->imageValidator = $imageValidator;
+        $this->configuration = $configuration;
+    }
+
+    public function assertFileUploadLimits(string $filePath): void
+    {
+        $size = filesize($filePath);
+        $maxProductImageSize = (new ByteConvertor(
+            (int) $this->configuration->get('PS_LIMIT_UPLOAD_IMAGE_VALUE'))
+        )
+            ->fromMegaByteToByte();
+
+        if ($maxProductImageSize > 0 && $size > $maxProductImageSize) {
```

Comment:
Does it mean if `$maxProductImageSize` is equal to zero we ignore the configuration and do not perform any check? The value can never be negative considering the form constraint, it can indeed be zero but then it can also be interpreted as purely disabling upload no? 🤷‍♂️

Reply:
I confess that I didn't think about this condition enough. Since I'm decorated ImageValidator with this class I simply copy this condition. What do you suggest exactly ? 🤔

Reply:
I don't think you need to decorate the initial `ImageValidator` since it's only used by two product handlers which need those new rules you can implement this new check directly in the existing class